biothings/web/query/builder.py

```diff
@@ -140,6 +146,54 @@ def __init__(
         # currently metadata is only used for __any__ query
         self.metadata = metadata
 
+    def _verify_default_regex_pattern(self, scopes_regexs):
+        """
+        We load the default annoation regex pattern from the settings and ensure it's
+        applied as the very last pattern in a the potential list of regex patterns provided
+        by the instance configuration
+
+        With the ANNOTATION_ID_REGEX_LIST configuration parameter, the user can provide
+        regex patterns matching the following structure:
+        (Union[str, re.Pattern], Union[str, Iterable])
+
+        We don't want to publically expose the default regex pattern in the configuration as
+        accidently modifying that could lead to unexpected / unwanted behavior. Therefore we
+        add it at runtime if it isn't discovered
+
+        Flow:
+        1) Branch on if a regex pattern list was provided. If none provided then set to the default
+        and return
+        2) If an iterable of regex patterns is provided then we force the structure into what we
+        expect: List[re.Pattern, Iterable]
+        3) We then iterate over the structure looking for the default regex pattern.
+            - If we find the default regex pattern match, we ignore updating our list
+            - If we don't find the default regex pattern match, we update our list with the pattern
+        4) At the end we add the default regex pattern because we've exhausted our search of the
+        current pattern list and trimmed any instances we found. This should ensure we've set the
+        default as the last instance in the regex pattern list
+        """
+        default_regex_pattern = ANNOTATION_DEFAULT_REGEX_PATTERN[0]
+        default_regex_fields = ANNOTATION_DEFAULT_REGEX_PATTERN[1]
```

Comment:
we should also allow `ANNOTATION_DEFAULT_REGEX_PATTERN` to be `None` if we want to disable it completely.

And even if it's not `None`, we should add some checks here to see if it's formatted correctly.